crates/test/src/rt/worker.rs

```diff
@@ -0,0 +1,66 @@
+//! Support for printing status information of a test suite in a browser.
+//!
+//! Currently this is quite simple, rendering the same as the console tests in
+//! node.js. Output here is rendered in a `pre`, however.
+
+use js_sys::Error;
+use wasm_bindgen::prelude::*;
+
+/// Implementation of `Formatter` for browsers.
+///
+/// Routes all output to a `pre` on the page currently. Eventually this probably
+/// wants to be a pretty table with colors and folding and whatnot.
+pub struct Worker {}
+
+#[wasm_bindgen]
+extern "C" {
+    type WorkerError;
+    #[wasm_bindgen(method, getter, structural)]
+    fn stack(this: &WorkerError) -> JsValue;
+
+    #[wasm_bindgen(js_name = "__wbg_test_output_writeln")]
+    fn write_output_line(data: JsValue);
+}
+
+impl Worker {
+    /// Attempts to create a new formatter for web worker
+    pub fn new() -> Worker {
+        Worker {}
+    }
+}
+
+impl super::Formatter for Worker {
+    fn writeln(&self, line: &str) {
+        write_output_line(JsValue::from(String::from(line)));
+    }
+
+    fn log_test(&self, name: &str, result: &Result<(), JsValue>) {
+        let s = if result.is_ok() { "ok" } else { "FAIL" };
+        self.writeln(&format!("test {} ... {}", name, s));
+    }
+
+    fn stringify_error(&self, err: &JsValue) -> String {
+        // TODO: this should be a checked cast to `Error`
+        let err = Error::from(err.clone());
+        let name = String::from(err.name());
+        let message = String::from(err.message());
+        let err = WorkerError::from(JsValue::from(err));
+        let stack = err.stack();
```

Comment:
Nit: you can avoid creating a JsValue when you already have one
```suggestion
        let error = Error::from(err.clone());
        let name = String::from(error.name());
        let message = String::from(error.message());
        let err = WorkerError::from(err);
        let stack = err.stack();
```

Reply:
`err` is behind a shared reference. `WorkerError` can not constructed from `&JsValue`. So we have to clone it twice.